src/components/Tx/SendTx/TxDescription.jsx

```diff
@@ -126,169 +120,234 @@ export default class TxDescription extends Component {
+              {gasError}
+            </StyledExecutionContextDetailsValue>
+          </StyledExecutionContextRow>
         )}
 
         {showTxExecutingFunction && (
-          <div className="execution-context__details-row">
-            <span className="execution-context__details-title">
+          <StyledExecutionContextRow>
+            <StyledExecutionContextDetailsTitle>
               {i18n.t('mist.sendTx.transactionExecutingFunction')}
-            </span>
-            <span className="execution-context__execution-function">
+            </StyledExecutionContextDetailsTitle>
+            <StyledExecutionContextExecutionFunction>
               {executionFunction.slice(0, executionFunction.indexOf('('))}
-            </span>
-          </div>
+            </StyledExecutionContextExecutionFunction>
+          </StyledExecutionContextRow>
         )}
 
-        <div className="execution-context__details-row">
-          <span className="execution-context__details-title">
+        <StyledExecutionContextRow>
+          <StyledExecutionContextTitle>
             {i18n.t('mist.sendTx.etherAmount')}
-          </span>
-          <span className="execution-context__details-value">
-            {this.formattedBalance(value)}
-          </span>
-        </div>
-
-        <div className="execution-context__details-row">
-          <span className="execution-context__details-title">
+          </StyledExecutionContextTitle>
+          <StyledExecutionContextDetailsValue>
+            {util.weiToEther(value).toString()}
+          </StyledExecutionContextDetailsValue>
+        </StyledExecutionContextRow>
+
+        <StyledExecutionContextRow>
+          <StyledExecutionContextTitle>
             {i18n.t('mist.sendTx.gasPrice')}
-          </span>
-          <span className="execution-context__details-value">{`${gweiPrice} GWEI`}</span>
-        </div>
+          </StyledExecutionContextTitle>
+          <StyledExecutionContextDetailsValue>{`${gasPriceGwei} gwei`}</StyledExecutionContextDetailsValue>
+        </StyledExecutionContextRow>
 
-        <div className="execution-context__details-row">
-          <span className="execution-context__details-title">
+        <StyledExecutionContextRow>
+          <StyledExecutionContextTitle>
             {i18n.t('mist.sendTx.gasEstimate')}
-          </span>
-          <span className="execution-context__details-value">{`${estimatedGas} WEI`}</span>
-        </div>
+          </StyledExecutionContextTitle>
+          <StyledExecutionContextDetailsValue>
+            {`${new BN(estimatedGas).toString()} gas`}
+          </StyledExecutionContextDetailsValue>
+        </StyledExecutionContextRow>
 
         {isTokenTransfer && (
           <div>
-            {tokenDisplayName && (
-              <div className="execution-context__details-row">
-                <span className="execution-context__details-title">
+            {tokenDisplayName && tokenDisplayName !== 'tokens' && (
```

Comment:
I think this component could use some breaking up. Could happen in a future PR if preferred.

Reply:
great idea, I just broke it up into smaller render functions 👌